Replicator/tests/ReplicatorAPITest.cc

```diff
@@ -613,6 +613,115 @@ TEST_CASE_METHOD(ReplicatorAPITest, "Is Document Pending", "[C][Push]") {
 }
 #endif
 
+#ifdef COUCHBASE_ENTERPRISE
+
+TEST_CASE_METHOD(ReplicatorAPITest, "Pending Document IDs Non-Existent Collection", "[C][Push]") {
+    ExpectingExceptions x;
+    // Create collection in the db and import documents
+    C4CollectionSpec Kyber = {"kyber"_sl, "crystal"_sl};
+    auto collKyber = createCollection(db, Kyber);
+    importJSONLines(sFixturesDir + "names_100.json", collKyber);
+
+    // Create the collection in the database to be replicated to
+    createDB2();
+    createCollection(db2, Kyber);
+
+    // Create collection spec but do not create collection in db
+    C4CollectionSpec Republic = {"republic"_sl, "galactic"_sl};
+
+    // Set options for the replication
+    C4Error err;
+    repl::C4ReplParamsCollection params{Kyber};
+    params.push = kC4OneShot;
+    params.pull = kC4Disabled;
+    params.callbackContext = this;
+    params.socketFactory = _socketFactory;
+
+    _repl = c4repl_newLocal(db, (C4Database*)db2, params, ERROR_INFO(err));
+    REQUIRE(_repl);
+
+    C4SliceResult encodedDocIDs = c4repl_getPendingDocIDs(_repl, Republic, &err);
+    CHECK(err.code == kC4ErrorNotOpen);
+    c4slice_free(encodedDocIDs);
+
+    c4repl_start(_repl, false);
+    REQUIRE_BEFORE(5s, c4repl_getStatus(_repl).level == kC4Stopped);
+    encodedDocIDs = c4repl_getPendingDocIDs(_repl, Republic, &err);
+    CHECK(err.code == kC4ErrorNotOpen);
+}
+
+TEST_CASE_METHOD(ReplicatorAPITest, "Pending Document IDs Multiple Collections", "[C][Push]") {
+    // Create collections and import documents to them
+    C4CollectionSpec Council = { "council"_sl, "jedi"_sl };
+    C4CollectionSpec Federation = { "federation"_sl, "trade"_sl };
+    auto collCouncil = createCollection(db, Council);
+    auto collFederation = createCollection(db, Federation);
+    importJSONLines(sFixturesDir + "names_100.json", collCouncil);
+    importJSONLines(sFixturesDir + "wikipedia_100.json", collFederation);
+
+    // Create the collections in the database to be replicated to
+    createDB2();
+    createCollection(db2, Council);
+    createCollection(db2, Federation);
+
+    // Set options for the replication
+    // Replicating only the Council collection
+    FLSliceResult options {};
+    C4Error err;
+    repl::C4ReplParamsCollection paramsCouncil { Council };
+    paramsCouncil.push = kC4OneShot;
+    paramsCouncil.pull = kC4Disabled;
+    paramsCouncil.callbackContext = this;
+    paramsCouncil.socketFactory = _socketFactory;
+
+    // Create replicator config for Federation collection
+    // This won't actually be replicated, but is needed to call c4repl_getPendingDocIDs for the collection
+    repl::C4ReplParamsCollection paramsFederation { Federation };
+    paramsFederation.push = kC4OneShot;
+    paramsFederation.pull = kC4Disabled;
+    paramsCouncil.callbackContext = this;
+    paramsCouncil.socketFactory = _socketFactory;
+
+    int expectedPending = 0;
+    SECTION("Normal") {
```

Comment:
Why is this SECTION as there is one? SECTIONs are for running two test scenarios in one test case.
This is a good case when only one collection is involved in each different replicators. Maybe to consider a case where two collections are included in one replicator

Reply:
Oh I see, again as above that was left over from when the test had multiple sections. I wasn't sure if it could be removed or not. Thank you 🙂